api/python/t4/packages.py

```diff
@@ -452,15 +459,26 @@ def fetch(self, dest='./'):
         """
         nice_dest = fix_url(dest).rstrip('/')
         file_list = []
-        
+        new_physical_keys = []
+
         for logical_key, entry in self.walk():
             logical_key = quote(logical_key)
             physical_key = _to_singleton(entry.physical_keys)
             new_physical_key = f'{nice_dest}/{logical_key}'
+            new_physical_keys.append(new_physical_key)
             file_list.append((physical_key, new_physical_key, entry.size, entry.meta))
 
         copy_file_list(file_list)
 
+        # return a package reroot package physical keys after the copy operation succeeds
+        # see GH#388 for context
+        pkg = Package()
+        for (logical_key, entry), physical_key in zip(self.walk(), new_physical_keys):
```

Comment:
Can you just do it in the loop above, in a single pass?

(But watch out for the quoted logical key. May want to rename the variable or something.)